File: src/2025-03-04-03-28-40--deepseek-chat-0304/javascript/exercises/practice/robot-name/robot-name.js
export class Robot {
  constructor() {
    this._name = null;
    this.generateName();
  }

  get name() {
    return this._name;
  }

  generateName() {
    let newName;
    do {
      // Generate two random letters (A-Z)
      const letters = String.fromCharCode(
        65 + Math.floor(Math.random() * 26),
        65 + Math.floor(Math.random() * 26)
      );
      // Generate three random digits
      const numbers = Math.floor(Math.random() * 1000)
        .toString()
        .padStart(3, '0');
      newName = letters + numbers;
    } while (Robot.usedNames.has(newName));
    
    this._name = newName;
    Robot.usedNames.add(newName);
  }

  reset() {
    const oldName = this._name;
    this.generateName();
    Robot.usedNames.delete(oldName);
  }
}

// Static property to track all used names
Robot.usedNames = new Set();

Robot.releaseNames = () => {
  Robot.usedNames.clear();
};

File: src/2025-03-04-03-28-40--deepseek-chat-0304/python/exercises/practice/robot-name/robot_name.py
import random
import string

class Robot:
    _used_names = set()

    def __init__(self):
        self._name = None

    def generate_name(self):
        while True:
            letters = ''.join(random.choices(string.ascii_uppercase, k=2))
            digits = ''.join(random.choices(string.digits, k=3))
            name = letters + digits
            if name not in self._used_names:
                self._used_names.add(name)
                return name

    @property
    def name(self):
        if not self._name:
            self._name = self.generate_name()
        return self._name

    def reset(self):
        if self._name:
            self._used_names.remove(self._name)
            self._name = None
            self._name = self.generate_name()
